lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -31,8 +31,17 @@ const (
 	zfsMaxVolBlocksize = 128 * 1024
 )
 
+func (d *zfs) isBlockBacking(vol Volume) bool {
+	return shared.IsTrue(vol.ExpandedConfig("zfs.block_mode"))
```

Comment:
This shouldn't use `ExpandedConfig` as we expect and require the block mode settings to be defined on the volume itself (i.e copied from the pool's setting at create time). So we should use `vol.Config()` instead.

Reply:
I've just noticed we have a `Volume.IsBlockBacked()` function.

I think we can use this, instead of this function, but it should be modified to return true of `block.filesystem` is in the volume's config.

I think this will work as a nice driver-agnostic way of detecting a block backed volume (combined with the existing logic you have to narrow down to only filesystem type volumes).

Also the existing use of `vol.IsBlockBacked()` in `validateVolumeCommonRules` could spell trouble for this PR (and makes for another reason to change it as described above), as a volume passed into it that isn't block backed may be incorrectly validated as if it were (based on the current pool's setting).